presto-spi/src/main/java/com/facebook/presto/spi/ColumnHandle.java

```diff
@@ -41,4 +42,13 @@ default ColumnHandle withRequiredSubfields(List<Subfield> subfields)
     {
         return this;
     }
+
+    /**
+     * Canonicalize domain of a column handle, by applying connector specific logic.
+     * This canonicalization will be used for history based optimizations.
+     */
+    default Domain canonicalizeDomain(Domain domain)
+    {
+        return domain;
+    }
```

Comment:
Do we need this in SPI? We are very careful changing SPI especially the underlying implementation is very specialized. We are only to compare `PARTITION_KEY`. So what' the downside of making `PARTITION_KEY` constant removed? It's anyway not needed right?

Reply:
Added javadoc for CONNECTOR strategy. I use it when canonicalizing constraint on tablescannodes. We remove constants for column domains when column is PARTITION_KEY type. 

I don't understand what you mean by its not needed

Reply:
I see your point now. CONNECTOR strategy will only be used for plan canonicalization It loses its meaning when we get to expressions/domains. 

Do you think there is a value in adding another enum for ExpressionCanonicalizationStrategy{DEFAULT, REMOVE_CONSTANTS} and passing it for expressions/domains?

Reply:
Let's chat in person on this. My hunch is that there is a way to avoid this interface.